### DISCUSSION TITLE
如何将已训练好的模型应用到流式服务器上

### DISCUSSION THREAD

**Discussion by phperfan** [PaddlePaddle/PaddleSpeech#2785] [General]
如题， 我已经按照 zh_en_tts/tts3 的教程训练好了模型， 命令行下也可以正常生成，
但是我想用流式tts的方式部署的时候， 提示不支持， 
麻烦问一下应该怎么部署， 或者有没有启动服务的相关代码借鉴，可以提供类似流式tts服务。

不好意思我对这个这块不太了解， 只是需要搭建一个类似的业务场景

**Comment by yt605155624** [comment]
zh_en_tts/tts3 默认提供的模型是非流式的模型，可以自己训练个中英文的流式模型，在这个 zh_en_tts/tts3 里面参考 https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/examples/csmsc/tts3/run_cnndecoder.sh 加个模型
然后在资源管理下新增你的模型文件
https://github.com/PaddlePaddle/PaddleSpeech/blob/96d76c83ade6dad94738ef4ae1a095dec3f858b3/paddlespeech/resource/pretrained_models.py#L1504
如果是中英文混合，还需要在服务代码里面加一下中英文混合的前端 https://github.com/PaddlePaddle/PaddleSpeech/blob/96d76c83ade6dad94738ef4ae1a095dec3f858b3/paddlespeech/server/engine/tts/online/onnx/tts_engine.py#L164

更多细节就需要你自己看代码了

ps 想实现流式不一定非要用服务，参考 https://github.com/PaddlePaddle/PaddleSpeech/blob/develop/paddlespeech/t2s/exps/stream_play_tts.py
